src/main/scala/scala/collection/next/NextIterableOnceOpsExtensions.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.collection
+package next
+
+private[next] final class NextIterableOnceOpsExtensions[A, CC[_], C](
+  private val col: IterableOnceOps[A, CC, C]
+) extends AnyVal {
+  /**
+   * Partitions this IterableOnce into a map according to a discriminator function `key`. All the values that
+   * have the same discriminator are then transformed by the `value` function and then reduced into a
+   * single value with the `reduce` function.
+   *
+   * {{{
+   *   def occurrences[A](as: IterableOnce[A]): Map[A, Int] =
+   *     as.iterator.groupMapReduce(identity)(_ => 1)(_ + _)
+   * }}}
+   *
+   * @note This will force the evaluation of the Iterator.
+   */
+  def groupMapReduce[K, B](key: A => K)(f: A => B)(reduce: (B, B) => B): immutable.Map[K, B] = {
+    val m = mutable.Map.empty[K, B]
+    col.foreach { elem =>
```

Comment:
I think we should use the exact implementation from the standard library—both for consistency, and to make it easier to apply changes from one to the other if needed

Reply:
Well, the implementation is largely the same on from the stdlib with some minor aesthetical changes which IMH, make the code easier to read.

But, I really do not have strong reasons for not changing them, if any other person believe I should make the change I would do it without problems.

Reply:
I think the stdlib likely doesn't use `foreach` for performance reasons, but I'm not sure. I'd rather the PR to make the code cleaner go to the stdlib first though, and then this library can follow suit, if that makes sense.

This is the only thing holding up this PR, to me

Reply:
@NthPortal AFAIK what is holding on this PR is #52 instead of this change or the structure proposals.

About this particular change, as I said I would like to hear what others think.

Reply:
I think we can merge this based on https://gitter.im/scala/contributors?at=60523e1d33346c16275ae3af —I think this PR is fairly uncontroversial. it's just moving a method up the collections hierarchy really

Reply:
@NthPortal so should I just push the revert of the style changes and this would be merged?
@SethTisue are you also ok with that _(both the merge and the code style change)_?

Reply:
@NthPortal btw I just realized this:

>I think the stdlib likely doesn't use foreach for performance reasons

Note that the stdlib does uses `foreach` just that using the `for` **sugar syntax**.
My change was just calling that method directly and using `updateWith` instead of `get` + **patten matching** + `put`
The performance should be the same, I just believe the code is cleaner now.

Reply:
my mistake. I thought the stdlib used a while loop for some reason